static/app/components/group/groupSummary.tsx

```diff
@@ -36,13 +37,29 @@ const makeGroupSummaryQueryKey = (
 
 export function useGroupSummary(groupId: string) {
   const organization = useOrganization();
-
-  return useApiQuery<GroupSummaryData>(
+  // We piggyback and use autofix's genai consent check for now.
+  const {
+    data: autofixSetupData,
+    isLoading: isAutofixSetupLoading,
+    isError: isAutofixSetupError,
+  } = useAutofixSetup({groupId});
+
+  const queryData = useApiQuery<GroupSummaryData>(
     makeGroupSummaryQueryKey(organization.slug, groupId),
     {
       staleTime: Infinity, // Cache the result indefinitely as it's unlikely to change if it's already computed
+      enabled: autofixSetupData?.genAIConsent.ok ?? false,
     }
   );
+
+  return {
+    ...queryData,
+    isLoading:
+      // isFetchedAfterMount is here so we don't have the split second where genAIConsent is true but the query isn't loading yet.
+      isAutofixSetupLoading || queryData.isLoading || !queryData.isFetchedAfterMount,
```

Comment:
Hmm are you sure that this is necessary? Seems like `isLoading` shouldn't switch to false when it gets enabled

Reply:
Huh, I just tried it: so `isLoading` is true even when a query is disabled? that seems weird.

Reply:
Yeah `isLoading` just means that the query hasn't loaded. This actually changed in v5 but we haven't upgraded yet